### PR TITLE
Added diff2html template report

### PR DESCRIPTION
**Pull request by davelopez**
In addition to the diff raw text file, an HTML report is generated with a more friendly visual representation of the differences.

Any improvement suggestions and ideas are greatly appreciated :)

### CODE REVIEW

File: tools/diff/diff.xml [hunk 1]
```diff
@@ -12,40 +13,93 @@
     </stdio>
 
     <command ><![CDATA[
-        diff -u '$input1' '$input2' > '$output1'
+        diff -u '$input1' '$input2' > '$diff_file';
+        #if $generate_report:
+            sed -e 's/@@title@@/Diff report for $input1.name and $input2.name/g'
```
Comment (bgruening) [inline tools/diff/diff.xml:18]: $input1.name --> use $input.element_identifier this should also work for collections. And please put them into single quotes.

Comment (davelopez) [reply tools/diff/diff.xml]: Done! Thanks for the comments!
If you see anything else or have some more usability ideas, please let me know :)

File: tools/diff/diff.xml [hunk 2]
```diff
@@ -12,40 +13,92 @@
     </stdio>
 
     <command ><![CDATA[
-        diff -u '$input1' '$input2' > '$output1'
+        diff -u '$input1' '$input2' > '$diff_file';
+        #if $generate_report:
+            sed -e 's/@@title@@/Diff report for '$input1.element_identifier' and '$input2.element_identifier'/g'
```
Comment (bgruening) [inline tools/diff/diff.xml:18]: lets see if this passes now, are you not closing the outer `'` here?

File: tools/diff/diff.xml [hunk 3]
```diff
@@ -12,40 +13,93 @@
...
+        #end if
+        
     ]]>
     </command>
 
     <inputs>
         <param name="input1" type="data" format="txt" label="First input file" />
         <param name="input2" type="data" format="txt" label="Second input file" />
+        <param name="generate_report" type="boolean" checked="true" truevalue="true" falsevalue="false" label="Generate HTML report" help="Generates an HTML report to visualize the differences"/>
+        <section name="adv" title="Advanced Report Options" expanded="false">
```
[the remark this answers is not in the document]
Comment (davelopez) [reply tools/diff/diff.xml:30]: Yeah, I got a bit carried away...

File: tools/diff/diff.xml (hunk 2 above)
Comment (davelopez) [reply tools/diff/diff.xml:18]: I thought the same, but... It works and passes the tests :)

**Comment by bgruening** [comment]
Cool, thanks a lot!

**Pull request merged**